osu.Game.Rulesets.Osu/Mods/OsuModAutopilot.cs

```diff
@@ -17,5 +23,41 @@ public class OsuModAutopilot : Mod
         public override string Description => @"Automatic cursor movement - just follow the rhythm.";
         public override double ScoreMultiplier => 1;
         public override Type[] IncompatibleMods => new[] { typeof(OsuModSpunOut), typeof(ModRelax), typeof(ModSuddenDeath), typeof(ModNoFail), typeof(ModAutoplay) };
+
+        public bool AllowFail => false;
+
+        private OsuInputManager inputManager;
+
+        private List<OsuReplayFrame> replayFrames;
+        private int frameIndex;
+
+        public void Update(Playfield playfield)
+        {
+            // If we are on the last replay frame, no need to do anything
+            if (frameIndex == replayFrames.Count - 1)
+            {
+                return;
+            }
+
+            // Check if we are closer to the next replay frame then the current one
+            if (Math.Abs(replayFrames[frameIndex].Time - playfield.Time.Current) >= Math.Abs(replayFrames[frameIndex + 1].Time - playfield.Time.Current))
```

Comment:
That would break the logic of the `if`: we only want to increment `frameIndex` if the condition is `true` (meaning that the time of the next frame is closer to the current time of the game then the time of the previous frame). Incrementing `frameIndex` each check would result in us moving to the next replay frame every update tick.